### PR TITLE
Check if org.opensuse.reference label contains CONTAINER_IMAGE_BUILD

### PR DESCRIPTION
**Pull request by pdostal**
- Related ticket: [poo#100916](https://progress.opensuse.org/issues/100916)
- Verification run: [SLE15-SP3-podman](http://pdostal-server.suse.cz/tests/87), [Ubuntu-22.04-docker](http://pdostal-server.suse.cz/tests/92), [RES8.5-docker](http://pdostal-server.suse.cz/tests/90) (unrelated error)

### CODE REVIEW

**Review by grisu48 — APPROVED** [review]
I think this can work like this.

We need to somehow ensure, that `jq` is available on all platforms we test BCI on. Unless someone knows this for sure, we need a lot of VRs here, basically one for each SLES version, plus RHEL plus Ubuntu

File: tests/containers/bci_prepare.pm [hunk 1]
```diff
@@ -106,6 +106,13 @@ sub run {
         record_info('IMAGE', $image);
         script_retry("$engine pull $image", timeout => 300, delay => 60, retry => 3);
         record_info('Inspect', script_output("$engine inspect $image"));
+
+        if (my $build = get_var('CONTAINER_IMAGE_BUILD')) {
+            record_info('CONTAINER_IMAGE_BUILD', $build) if ($build != 'UNKNOWN');
+            my $reference = script_output(qq($engine inspect --type image $image | jq -r '.[0].Config.Labels."org.opensuse.reference"'));
+            record_info('org.opensuse.reference', $reference);
+            die('We did not pull the correct image!') if ($build != 'UNKNOWN' && $reference !~ /$build$/);
```
Comment (jlausuch) [inline tests/containers/bci_prepare.pm:114]:
```suggestion
            die('Miss match in image build number. The image build number is different than the one triggered by the container bot') if ($build ne 'UNKNOWN' && $reference !~ /$build$/);
```

File: tests/containers/bci_prepare.pm [hunk 2]
```diff
@@ -106,6 +106,13 @@ sub run {
         record_info('IMAGE', $image);
         script_retry("$engine pull $image", timeout => 300, delay => 60, retry => 3);
         record_info('Inspect', script_output("$engine inspect $image"));
+
+        if (my $build = get_var('CONTAINER_IMAGE_BUILD')) {
+            record_info('CONTAINER_IMAGE_BUILD', $build) if ($build != 'UNKNOWN');
```
[the remark this answers is not in the document]
Comment (jlausuch) [reply tests/containers/bci_prepare.pm:111]: You can also put all this code inside a condition 
```
if ($build ne 'UNKNOWN') {
    record_info...
    my $reference = ...
    record_info...
    die(...)
}

[the remark this answers is not in the document]
Comment (pdostal) [reply tests/containers/bci_prepare.pm]: Solved and verified by http://pdostal-server.suse.cz/tests/94.

File: tests/containers/bci_prepare.pm [hunk 3]
```diff
@@ -106,6 +106,14 @@ sub run {
         record_info('IMAGE', $image);
         script_retry("$engine pull $image", timeout => 300, delay => 60, retry => 3);
         record_info('Inspect', script_output("$engine inspect $image"));
+
+        my $build = get_var('CONTAINER_IMAGE_BUILD');
+        if ($build && $build ne 'UNKNOWN') {
+            record_info('CONTAINER_IMAGE_BUILD', $build);
```
Comment (jlausuch) [inline tests/containers/bci_prepare.pm:112]: Maybe this info is superfluous, we can remove it, it's already in the variable. 
Don't like much how this text fits in the frame:
image

File: tests/containers/bci_prepare.pm [hunk 4]
```diff
@@ -106,6 +106,14 @@ sub run {
...
+            my $reference = script_output(qq($engine inspect --type image $image | jq -r '.[0].Config.Labels."org.opensuse.reference"'));
+            record_info('org.opensuse.reference', $reference);
```
Comment (jlausuch) [inline tests/containers/bci_prepare.pm:114]: Similar here, 
image

but however, this `record_info` is useful.

```suggestion
            record_info('Image ref', "org.opensuse.reference = $reference");
```

File: tests/containers/bci_prepare.pm [hunk 5]
```diff
@@ -106,6 +106,14 @@ sub run {
...
+            die('Miss match in image build number. The image build number is different than the one triggered by the container bot!') if ($reference !~ /$build$/);
```
Comment (jlausuch) [inline tests/containers/bci_prepare.pm:115]: if we skip the first `record_info`, we could do:
```suggestion
            die("Miss match in image build number. The image build number extracted from the image ($reference) is different than the one triggered by the container bot ($build)!") if ($reference !~ /$build$/);
```

Comment (pdostal) [reply tests/containers/bci_prepare.pm:115]: Please not. I also want to see the image ref and the build number when this check is passing.

File: tests/containers/bci_prepare.pm (hunk 4 above)
Comment (pdostal) [reply tests/containers/bci_prepare.pm:114]: I did:
```suggestion
            record_info('image ref', "org.opensuse.reference: $reference");
```

File: tests/containers/bci_prepare.pm (hunk 3 above)
Comment (pdostal) [reply tests/containers/bci_prepare.pm:112]: I did:
```suggestion
            record_info('BUILD#', "CONTAINER_IMAGE_BUILD=$build");
```

**Review by jlausuch — APPROVED** [review]
LGTM!

**Pull request merged**